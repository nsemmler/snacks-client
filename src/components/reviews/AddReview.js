import React from 'react';
import {
  Button,
  Col,
  Container,
  Form,
  FormGroup,
  Label,
  Input
} from 'reactstrap';
import ReactStars from 'react-stars'

class AddReview extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      reviewRate: 0
    }
  }

  ratingChanged = (newRating) => {
    this.setState({reviewRate: newRating})
  }

  render() {
    return (<Container>
      <Form className="form-horizontal well" onSubmit={event => {
          event.preventDefault();
          this.props.handleAddReview(event.target.title.value, event.target.text.value, this.state.reviewRate);
          this.props.toggleAdd();
        }}>
        <div>
          <h4>Add Review</h4>
        </div>
        <FormGroup>
          <Label for="rating" sm={2}>
            Rating:</Label>
          <ReactStars count={5} value={this.state.reviewRate} onChange={this.ratingChanged} half={false} size={24} color2={'#ffd700'} required="required"/>
        </FormGroup>

        <FormGroup row="row">
          <Label for="title" sm={2}>Title</Label>
          <Col sm={9}>
            <Input type="textarea" required="required" name="title" id="title"/>
          </Col>
        </FormGroup>

        <FormGroup row="row">
          <Label for="description" sm={2}>Description</Label>
          <Col sm={9}>
            <Input type="textarea" required="required" name="text" id="text"/>
          </Col>
        </FormGroup>
        <FormGroup check="check" row="row">
          <Col sm={{
              size: 10,
              offset: 2
            }}>
            <Button outline="outline" color="success">Submit</Button>
          </Col>
        </FormGroup>
      </Form>
    </Container>);
  }
}

export default AddReview;
