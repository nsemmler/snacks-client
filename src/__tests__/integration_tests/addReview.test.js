import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import AddReview from '../../components/reviews/AddReview'

describe('AddReview Component', () => {
  test('renders consistently', () => {
    const form = renderer.create(
      <MemoryRouter>
        <AddReview />
      </MemoryRouter>
    ).toJSON()
    expect(form).toMatchSnapshot()
  })
})
