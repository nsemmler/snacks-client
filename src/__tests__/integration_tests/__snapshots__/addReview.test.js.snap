// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddReview Component renders consistently 1`] = `
<div
  className="container"
>
  <form
    className="form-horizontal well"
    onSubmit={[Function]}
  >
    <div>
      <h4>
        Add Review
      </h4>
    </div>
    <div
      className="position-relative form-group"
    >
      <label
        className="col-sm-2 col-form-label"
        htmlFor="rating"
      >
        Rating:
      </label>
      <ul
        className="rc-rate"
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseLeave={[Function]}
        role="radiogroup"
        style={Object {}}
        tabIndex={0}
      >
        <li
          aria-checked="false"
          aria-posinset={1}
          aria-setsize={5}
          className="rc-rate-star rc-rate-star-zero"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseMove={[Function]}
          role="radio"
          tabIndex={0}
        >
          <div
            className="rc-rate-star-first"
          >
            ★
          </div>
          <div
            className="rc-rate-star-second"
          >
            ★
          </div>
        </li>
        <li
          aria-checked="false"
          aria-posinset={2}
          aria-setsize={5}
          className="rc-rate-star rc-rate-star-zero"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseMove={[Function]}
          role="radio"
          tabIndex={0}
        >
          <div
            className="rc-rate-star-first"
          >
            ★
          </div>
          <div
            className="rc-rate-star-second"
          >
            ★
          </div>
        </li>
        <li
          aria-checked="false"
          aria-posinset={3}
          aria-setsize={5}
          className="rc-rate-star rc-rate-star-zero"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseMove={[Function]}
          role="radio"
          tabIndex={0}
        >
          <div
            className="rc-rate-star-first"
          >
            ★
          </div>
          <div
            className="rc-rate-star-second"
          >
            ★
          </div>
        </li>
        <li
          aria-checked="false"
          aria-posinset={4}
          aria-setsize={5}
          className="rc-rate-star rc-rate-star-zero"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseMove={[Function]}
          role="radio"
          tabIndex={0}
        >
          <div
            className="rc-rate-star-first"
          >
            ★
          </div>
          <div
            className="rc-rate-star-second"
          >
            ★
          </div>
        </li>
        <li
          aria-checked="false"
          aria-posinset={5}
          aria-setsize={5}
          className="rc-rate-star rc-rate-star-zero"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseMove={[Function]}
          role="radio"
          tabIndex={0}
        >
          <div
            className="rc-rate-star-first"
          >
            ★
          </div>
          <div
            className="rc-rate-star-second"
          >
            ★
          </div>
        </li>
      </ul>
    </div>
    <div
      className="position-relative row form-group"
    >
      <label
        className="col-sm-2 col-form-label"
        htmlFor="title"
      >
        Title
      </label>
      <div
        className="col-sm-9"
      >
        <textarea
          className="form-control"
          id="title"
          name="title"
          required="required"
        />
      </div>
    </div>
    <div
      className="position-relative row form-group"
    >
      <label
        className="col-sm-2 col-form-label"
        htmlFor="description"
      >
        Description
      </label>
      <div
        className="col-sm-9"
      >
        <textarea
          className="form-control"
          id="text"
          name="text"
          required="required"
        />
      </div>
    </div>
    <div
      className="position-relative row form-check"
    >
      <div
        className="col-sm-10 offset-sm-2"
      >
        <button
          className="btn btn-outline-success"
          onClick={[Function]}
          type={undefined}
        >
          Submit
        </button>
      </div>
    </div>
  </form>
</div>
`;
